Tella/Data/Database/TellaDataBase.swift

```diff
@@ -591,3 +592,89 @@ class TellaDataBase {
     }
     
 }
+
+extension TellaDataBase {
+    
+    func createFeedbackTable() {
+        
+        let columns = [
+            cddl(D.cId, D.integer, primaryKey: true, autoIncrement: true),
+            cddl(D.ctext, D.text),
+            cddl(D.cStatus, D.integer),
+            cddl(D.cCreatedDate, D.float),
+            cddl(D.cUpdatedDate, D.float),
+        ]
+        statementBuilder.createTable(tableName: D.tFeedback, columns: columns)
+        
+    }
+    
+    func getDraftFeedback() -> Feedback? {
+        do {
+            let feedbackCondition = [KeyValue(key: D.cStatus, value: FeedbackStatus.draft.rawValue)]
+
+            let feedbackDict = try statementBuilder.getSelectQuery(tableName: D.tFeedback,
+                                                                   equalCondition: feedbackCondition)
+            return try feedbackDict.first?.decode(Feedback.self)
+        } catch {
+            return nil
+        }
+    }
+    
+    func getUnsentFeedbacks() -> [Feedback] {
+        do {
+            let feedbackCondition = [KeyValue(key: D.cStatus, value: FeedbackStatus.pending.rawValue)]
+
+            let feedbackDict = try statementBuilder.getSelectQuery(tableName: D.tFeedback,
+                                                                   equalCondition: feedbackCondition)
+            return try feedbackDict.decode(Feedback.self)
+        } catch {
+            return []
+        }
+    }
+
+    func addFeedback(feedback : Feedback) -> Int? {
+        do {
+            let valuesToAdd = [KeyValue(key: D.ctext, value: feedback.text),
+                               KeyValue(key: D.cStatus, value: feedback.status?.rawValue),
+                               KeyValue(key: D.cCreatedDate, value: Date().getDateDouble()),
+                               KeyValue(key: D.cUpdatedDate, value: Date().getDateDouble())]
+            
+           return try statementBuilder.insertInto(tableName: D.tFeedback,
+                                            keyValue: valuesToAdd)
+            
+        } catch(let error) {
+            debugLog(error)
+            return nil
+        }
+    }
+    
+    func updateFeedback(feedback : Feedback) {
+        
+        do {
+            
+            let valuesToUpdate = [ KeyValue(key: D.ctext, value: feedback.text),
+                                   KeyValue(key: D.cStatus, value: feedback.status?.rawValue),
+                                  KeyValue(key: D.cUpdatedDate, value: Date().getDateDouble())]
+            
+            let feedbackCondition = [KeyValue(key: D.cId, value: feedback.id)]
+            try statementBuilder.update(tableName: D.tFeedback,
+                                        keyValue: valuesToUpdate,
+                                        primarykeyValue: feedbackCondition)
+            
+        } catch(let error) {
+            debugLog(error)
+        }
+        
+    }
+    
+    
+    func deleteFeedback(feedbackId: Int?) {
```

Comment:
here we are sending feedbackId as optional so I supposed if the feedbackId is nil the query will crash right? Probably we should safely unwrap optional values before executing queries using if let or guard let.

```
guard let feedbackId = feedbackId else {
            return 
        }
 ```

Reply:
This won't crash but it's better to unwrap optional values before executing queries